src/f32-rsum/rvv.c.in

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Imagination Technologies, Inc.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$assert LMUL in [1, 2, 4, 8]
+#include <assert.h>
+
+#include <riscv_vector.h>
+#include "xnnpack/common.h"
+#include "xnnpack/reduce.h"
+
+void xnn_f32_rsum_ukernel__rvv_u${LMUL}v(
+    size_t batch,
+    const float* input,
+    float* output,
+    const struct xnn_f32_scaleminmax_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  const float scale = params->scalar.scale;
+  const float min = params->scalar.min;
+  const float max = params->scalar.max;
+
+  batch >>= XNN_LOG2_SIZEOF_FLOAT;
+  vfloat32m1_t acc_f32v = __riscv_vfmv_s_f_f32m1(0.f, __riscv_vsetvl_e32m1(batch));
+  $if LMUL == 1 or LMUL == 2:
+    size_t n = __riscv_vsetvl_e32m${LMUL}(batch);
+    vfloat32m${LMUL}_t sum0_f32v = __riscv_vfmv_v_f_f32m${LMUL}(0.f, n);
+    vfloat32m${LMUL}_t sum1_f32v = __riscv_vfmv_v_f_f32m${LMUL}(0.f, n);
+    for (; batch >= n * 8; batch -= n * 8) {
+      vfloat32m${LMUL}_t in0_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in1_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in2_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in3_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in4_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in5_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in6_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
+      vfloat32m${LMUL}_t in7_f32v = __riscv_vle32_v_f32m${LMUL}(input, n); input += n;
```

Comment:
unrolling is unconventional for RVV... we use LMUL to achieve that?
if LMUL is large, unrolling will run out of registers.